### DISCUSSION TITLE
[riverpod_generator] Refactoring provider class with riverpod_generator

### DISCUSSION THREAD

**Discussion by samanzamani** [rrousselGit/riverpod#2278] [Q&A] [answered]
Hi.
I want to refactor my old provider class with `riverpod_generator` and I am slightly confused.
I have the `AppProvider` class below, which provides App-related things like `route` for me.

```dart
class AppProvider {
  static final sharedPreferences =
      Provider<SharedPreferences>((_) => throw UnimplementedError());

  static final auth = Provider<AuthService>((ref) {
    return AuthService.init(sharedPreference: ref.watch(sharedPreferences));
  });

  static final route = Provider<GoRouter>((ref) {
    final authService = ref.watch(AppProvider.auth);
    return useMemoized(
      () => appRouter(auth: authService),
    );
  });

  static final theme = Provider<AppTheme>(
    (ref) {
      return AppTheme.app();
    },
  );
}

```

Everywhere I need them I used `final appRouter = ref.watch(AppProvider.route);` code.

How can I use `riverpod_generator` to refactor this class? Is it a good idea to use `riverpod_generator` in this case?

**Comment by rrousselGit** [comment, marked as answer]
`Static final ` providers aren't supported. You may have to remove this namespace class.

I'd suggest using import alias for this:

```dart
// app_provider.dart
@rvierpod
SharedPref example(...) => ...;

// main.dart
import 'app_provider.dart' as app_provider;
...
ref.watch(app_provider.exampleProvider);
```
